### PR TITLE
Validate IP with DNS, show when they don't match

### PR DESCRIPTION
**Pull request by sbeaulie**
Also adding a creation_timestamp as found in the annotations when created by vmpooler

### CODE REVIEW

File: lib/vmstatus/list.rb [hunk 1]
```diff
@@ -38,26 +39,36 @@ def output(results)
...
               end
 
-      hostname = vm.hostname.length > 15 ? (vm.hostname[0..12] + "...") : vm.hostname
-      on = vm.on? ? 'on' : 'off'
-      running = vm.running? ? '*' : '!'
-
-      checkout = vm.checkout ? vm.checkout.to_s : ' '
-      ttl = vm.ttl ? ("%8.2fh" % vm.ttl) : 'never'
-      user = vm.user ? vm.user : ' '
+      hostname = vm.hostname.length > 24 ? (vm.hostname[0..21] + "...") : vm.hostname
```
Comment (genebean) [inline lib/vmstatus/list.rb:42]: why are we truncating hostnames and why change the current value?

File: lib/vmstatus/list.rb [hunk 2]
```diff
@@ -38,26 +39,36 @@ def output(results)
...
+      # maps the vmpooler host to CI type, redis used to run on the same host as the vmpooler application.
       pooler = case vm.vmpooler
                when 'vmpooler'
                  'ci-old'
                when 'vmpooler-cinext'
                  'ci-next'
                when 'vmpooler-dev'
                  'ci-dev'
+               when 'vmpooler-redis-prod-2.delivery.puppetlabs.net'
+                 'ci-next'
```
Comment (genebean) [inline lib/vmstatus/list.rb:59]: Maybe a task for later, but I assume this needs updating for `vmpooler-prod` and `vmpooler-stage` in k8s too.

File: lib/vmstatus/summary.rb [hunk 3]
```diff
@@ -71,7 +88,19 @@ def output(results)
...
-            statsd.gauge("#{vmpooler}.#{state}", count)
+            pooler_short = case vmpooler
+                     when 'vmpooler'
+                       'ci-old'
+                     when 'vmpooler-cinext'
+                       'ci-next'
+                     when 'vmpooler-dev'
+                       'ci-dev'
+                     when 'vmpooler-redis-prod-2.delivery.puppetlabs.net'
+                       'ci-next'
```
Comment (genebean) [inline lib/vmstatus/summary.rb:99]: Same note as above re k8s

File: lib/vmstatus/list.rb (hunk 1 above)
Comment (sbeaulie) [reply lib/vmstatus/list.rb:42]: good point some VMs have longer names when created manually

File: lib/vmstatus/list.rb (hunk 2 above)
Comment (sbeaulie) [reply lib/vmstatus/list.rb:59]: yes we should. Its for statsd though like what the prefix we should use? Nevermind I think its for display in the columns.

**Comment by sbeaulie** [comment]
Here is an example for the output
```
Hostname                 IP(vm)          IP(dns)         Cluster Host                                 Type                   Pooler  Status    Running  Creation Time Vcenter    Checkout Time                     TTL User                 Jenkins Job
andean-mystery           10.16.113.177   *N/A            pix-jj28-u19.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/!    2020-12-17 23:45:49 UTC  2020-12-17T23:52:45+00:00      64.82h molly.waggett        
bare-pensioner           10.16.126.125   *N/A            pix-jj29-u21.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/!    2020-12-14 20:46:52 UTC  2020-12-14T22:21:00+00:00     103.29h nick.burgan-illig    
clammy-rival             10.16.114.129   10.16.114.129   pix-jj27-u19.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-23 06:03:56 UTC  2020-12-23T06:19:57+00:00       3.27h maheswaran           
coherent-berth           10.16.112.197   10.16.112.197   pix-jj27-u19.ops.puppetlabs.net              redhat-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-22 22:02:34 UTC  2020-12-23T00:51:44+00:00      57.82h justin               
elder-saturday           10.16.119.186   10.16.119.186   pix-jj28-u20.ops.puppetlabs.net              ubuntu-1804-x86_64-... ci-next adhoc      on/*    2020-12-15 14:05:18 UTC  2020-12-15T16:13:01+00:00     109.17h                      
evasive-amorist          10.16.118.82    10.16.118.82    pix-jj29-u20.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-22 22:02:32 UTC  2020-12-23T03:28:32+00:00       0.42h maheswaran           
foul-narrowing           10.16.127.105   *N/A            pix-jj27-u20.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/!    2020-12-14 20:46:52 UTC  2020-12-14T22:21:00+00:00     103.29h nick.burgan-illig    
great-dissenter          10.16.120.15    10.16.120.15    pix-jj27-u20.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-17 22:23:28 UTC  2020-12-17T23:45:46+00:00      76.71h nick.burgan-illig    
inclusive-mien           10.16.125.21    10.16.125.21    pix-jj27-u22.ops.puppetlabs.net              win-2016-x86_64-pixa4  ci-next adhoc      on/*    2020-12-18 17:59:28 UTC  2020-12-18T18:01:04+00:00      94.97h belen.bustamante     
kinder-stupor            10.16.124.213   *N/A            pix-jj28-u22.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/!    2020-12-17 22:23:28 UTC  2020-12-17T23:45:46+00:00      76.71h nick.burgan-illig    
mean-disservice          10.16.113.203   *N/A            pix-jj29-u21.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/!    2020-12-17 22:23:28 UTC  2020-12-17T23:45:46+00:00      76.70h nick.burgan-illig    
numb-crowbait            10.16.114.184   10.16.114.184   pix-jj29-u21.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-15 01:16:15 UTC  2020-12-15T02:00:44+00:00      94.95h molly.waggett        
only-neutrality          10.16.114.218   *N/A            pix-jj29-u19.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/!    2020-12-18 17:44:00 UTC  2020-12-18T17:58:37+00:00     194.92h nick.burgan-illig    
pyhrric-phrase           10.16.116.149   10.16.116.149   pix-jj27-u19.ops.puppetlabs.net              win-2016-x86_64-pixa4  ci-next adhoc      on/*    2020-12-23 09:09:20 UTC  2020-12-23T09:10:55+00:00       6.13h selvakumar.subburaj  
slow-ruffian             10.16.124.71    10.16.124.71    pix-jj29-u22.ops.puppetlabs.net              redhat-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-22 17:00:22 UTC  2020-12-23T00:04:24+00:00      57.03h justin               
stormy-flight            10.16.126.205   10.16.126.205   pix-jj27-u21.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-14 21:52:43 UTC  2020-12-14T22:21:00+00:00     103.29h nick.burgan-illig    
virtual-gibbet           10.16.117.204   10.16.117.204   pix-jj29-u22.ops.puppetlabs.net              ubuntu-1804-x86_64-... ci-next adhoc      on/*    2020-12-13 21:05:31 UTC  2020-12-13T21:05:51+00:00      66.05h dorin.pleava         
waxy-transistor          10.16.117.207   10.16.117.207   pix-jj28-u19.ops.puppetlabs.net              redhat-7-x86_64-pixa4  ci-next adhoc      on/*    2020-12-23 09:32:47 UTC  2020-12-23T10:42:25+00:00       7.66h ioannis.karasavvaidis 
abrupt-hubby             10.16.122.92    10.16.122.92    pix-jj27-u21.ops.puppetlabs.net              centos-7-x86_64-pixa4  ci-next building   on/*    2020-12-23 14:06:03 UTC  2020-12-23T14:06:44+00:00      11.05h always-be-scheduling enterprise_pe-acceptance-tests_integration-system_pe_full-upgrade_nightly_2019.8.x
abrupt-kidding           10.16.118.220   10.16.118.220   pix-jj29-u22.ops.puppetlabs.net              centos-8-x86_64-pixa4  ci-next building   on/*    2020-12-23 14:09:52 UTC  2020-12-23T14:10:20+00:00      11.13h always-be-scheduling enterprise_pe-acceptance-tests_integration-system_pe_ha_failover_nightly_2019.8.x
absolute-gusto           10.16.120.211   10.16.120.211   pix-jj26-chassis1-3.ops.puppetlabs.net       sles-12-x86_64         ci-next building   on/*    2020-12-23 14:07:04 UTC  2020-12-23T14:07:49+00:00      11.09h always-be-scheduling enterprise_pe-acceptance-tests_integration-system_pe_ha-upgrade_nightly_2019.8.x
absurd-subtype           10.16.120.22    10.16.120.22    pix-jj26-chassis1-7.ops.puppetlabs.net       centos-7-x86_64        ci-next building   on/*    2020-12-23 14:05:01 UTC  2020-12-23T14:05:44+00:00      11.03h always-be-scheduling enterprise_pe-acceptance-tests_integration-system_pe_lei-split-upgrade_nightly_2019.8.x
```

**Comment by genebean** [comment]
Should the Jenkins Job column also get an `N/A' entry?

**Pull request closed without merging**